test/distributed/org/apache/cassandra/distributed/test/sai/SlowSAIQueryLoggerTest.java

```diff
@@ -268,6 +269,30 @@ public void testSlowSAIQueryLogger() throws Throwable
                               "SAI slowest query plan:",
                               "LiteralIndexScan");
 
+            // Disable query optimizer to prevent skipping hybrid query logic and hit orderByResults to verify metrics update
+            node.runOnInstance(() -> QueryController.QUERY_OPT_LEVEL = 0);
```

Comment:
Nit: maybe we should reset the optimization level after this check, so the remaining checks work with defaults?
```java
node.runOnInstance(() -> QueryController.QUERY_OPT_LEVEL = CassandraRelevantProperties.SAI_QUERY_OPTIMIZATION_LEVEL.getInt());
```